src/encoder.rs

```diff
@@ -1475,34 +1475,30 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
         Ok(())
     }
 
-    fn write_frame_lrf(&mut self, seq: &Sequence, fi: &FrameInvariants) -> io::Result<()> {
+    fn write_frame_lrf(&mut self, seq: &Sequence, fi: &FrameInvariants,
+                       rs: &RestorationState) -> io::Result<()> {
       if seq.enable_restoration && !fi.allow_intrabc { // && !self.lossless
         let mut use_lrf = false;
         let mut use_chroma_lrf = false;
         for i in 0..PLANES {
-          self.write(2,fi.lrf_types[i])?; // filter type by plane
-          if fi.lrf_types[i] != RESTORE_NONE {
+          self.write(2,rs.lrf_type[i])?; // filter type by plane
+          if rs.lrf_type[i] != RESTORE_NONE {
             use_lrf = true;
             if i > 0 { use_chroma_lrf = true; }
           }
         }
         if use_lrf {
-          // At present, we're locked to a restoration unit size equal to superblock size.
-          // Signal as such.
-          if seq.use_128x128_superblock {
-            self.write(1,0)?; // do not double the restoration unit from 128x128
-          } else {
-            self.write(1,0)?; // do not double the restoration unit from 64x64
+          // The Y shift value written here indicates shift up from superblock size
+          if !seq.use_128x128_superblock {
+            self.write(1, if rs.unit_size[0] > 64 {1} else {0})?;
+          }
+          if rs.unit_size[0] > 64 {
+            self.write(1, if rs.unit_size[0] > 128 {1} else {0})?;
```

Comment:
Is it possible to have both conditions above true at the same time?

Reply:
Er, I should clarify-- yes, it's possible.  Previous code was simply hardwiring the case we were using, I decided I to replace it with the general case.